packages/ensnode-react/src/hooks/useAvatarUrl.ts

```diff
@@ -20,6 +21,111 @@ import { useRecords } from "./useRecords";
  */
 const AVATAR_TEXT_RECORD_KEY = "avatar" as const;
 
+/**
+ * Resolves an avatar text record to a browser-supported URL.
+ * This is the core resolution logic extracted for testing.
+ *
+ * @param avatarTextRecord - The raw avatar text record value from ENS
+ * @param name - The ENS name
+ * @param namespaceId - The ENS namespace ID
+ * @param browserSupportedAvatarUrlProxy - Optional custom proxy function
+ * @returns The resolved avatar URL result
+ * @internal
+ */
+export function resolveAvatarUrl(
```

Comment:
```suggestion
export function buildBrowserSupportedAvatarUrl(
```

Goal: We shouldn't use "resolve" terminology here. In ENS, "resolve" has special meaning. For example, you can say "resolve the avatar text record for the name X". But note how the signature of this function requires that you already have completed a resolution of the avatar text record.

Please also note in the suggested function name I wrote "...AvatarUrl" and not "AssetUrl". That was intentional.